### ISSUE TITLE
Failure while “make”:  /usr/bin/ld: failed to set dynamic section sizes: bad value

### ISSUE THREAD

**Issue by songyahui** [snort3/snort3#320]
Hi, I keep getting errors when executing make. Here are the error messages. 
I am using Ubuntu 22.04.2 LTS.

```
Consolidate compiler generated dependencies of target side_channel
[ 86%] Built target side_channel
Consolidate compiler generated dependencies of target connectors
[ 86%] Built target connectors
Consolidate compiler generated dependencies of target file_connector
[ 86%] Built target file_connector
Consolidate compiler generated dependencies of target snort
[ 86%] Linking CXX executable snort
/usr/bin/ld: /usr/local/lib/libdaq_static_afpacket.a(libdaq_static_afpacket_la-daq_afpacket.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIE
/usr/bin/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/snort.dir/build.make:1767: src/snort] Error 1
make[1]: *** [CMakeFiles/Makefile2:2795: src/CMakeFiles/snort.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

**Comment by ansrb** [comment]
A line above the one in the title suggests rebuilding DAQ with -fPIE flag.
When configuring both DAQ and snort with default compiling options it should work fine. Did you provide any special compiling options? If you did, please share your configuration for building, `./configure` call for DAQ and `./configure_cmake.sh` call for snort

**Comment by thiagocrepaldi** [comment]
I am having the same issue using latest onnx main branch, no changes at all

**Comment by ssam18** [comment]
Yes, this is still reproducible on Ubuntu 22.04+ with a default ./configure && make install of libdaq. The root cause isn't really snort, it's that libdaq's static module archives (libdaq_static_afpacket.a etc.) get built without -fPIC, and modern Ubuntu/Debian/Fedora toolchains default to PIE executables, so the link breaks. The misleading bit is the linker says "recompile with -fPIE" but the actual fix is to rebuild libdaq with -fPIC.

Quick workaround: rebuild libdaq with ./configure --with-pic CFLAGS=-fPIC && make && sudo make install, or configure snort with --disable-static-daq to use dynamic modules instead.

I've opened two PRs to fix:
1. snort3/libdaq#39 makes libdaq build PIC static archives by default, and 
2. snort3/snort3#462 adds a configure-time check in snort that fails fast with a clear message instead of letting people hit the 
cryptic ld error twenty minutes into the build.
